### PR TITLE
publiccloud: Use serial_terminal to setup PC instance 

### PR DESCRIPTION
**Pull request by cfconrad**
This can speedup a usual extramau testrun from ~60min to ~30min.

Behavior:
  The function `select_host_console()` gives always a serial_terminal
  console on the **host** (not the Public Cloud instance).
  If this function is called, while TUNNELED is already setup, we die()
  unless the `force=>1` parameter is given, which kill the TUNNELED
  session.

  The trick is, that we call `select_serial_teriminal()` with `TUNNELED
  == 0` and do the `ssh -ty` hook after the preconfiguration is done.

- Related ticket: ???
- Verification run: ~https://openqa.suse.de/tests/5779825~
  - https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12236#issuecomment-816155083

### CODE REVIEW

File: tests/publiccloud/download_repos.pm [hunk 1]
```diff
@@ -30,7 +30,7 @@ sub get_repo_status {
 sub run {
     my ($self, $args) = @_;
 
-    select_host_console();    # select console on the host, not the PC instance
+    $self->select_serial_terminal();
```
Comment (cfconrad) [inline tests/publiccloud/download_repos.pm:33]: Maybe we should think of using `serial_terminal()` always for `host_console()` ?! WDYT?

**Comment by cfconrad** [comment]
Same test without `CASEDIR` https://openqa.suse.de/tests/5779837

I expect a duration ~24min

**Review by okurz — CHANGES_REQUESTED** [review]
I assume the commit message should say "publiccloud: Use …"

And also s/This speedup/This speeds up/

File: tests/publiccloud/download_repos.pm (hunk 1 above)
Comment (cfconrad) [reply tests/publiccloud/download_repos.pm:33]: AFAIK, the `root-console` and `user-console` are both "graphical" consoles. What I'm using is the serial terminal.

check https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L1190 it falls back to `root-console` but if `VIRTIO_CONSOLE` or some other faster console is available, it takes it.

So I agree, we would need to double check, which tests can be changed to something faster. But in general for text only tests I would recommend to use the `serial_terminal`.

**Review by okurz — APPROVED** [review]

**Review by grisu48 — APPROVED** [review]
LGTM!

**Comment by cfconrad** [comment]
@pdostal did a VR https://openqa.suse.de/tests/5785828 and it failed with 
```
# Test died: Console 'root-virtio-terminal' activated in TUNNEL mode activated but tunnel(s) are not yet initialized, use the 'tunnel' console and call 'setup_ssh_tunnels' first
```
Looks like I cloned the wrong job, which do not use `TUNNELED` :/

---

From what I understand, the `TUNNELED` consoles are all `tty` one. These where we use the "serial-terminal" to get the exit codes. Which we do by a pseudo serial file via SSH using some magic pipes :rocket: in `TUNNELED` mode.

The sanity check https://github.com/os-autoinst/os-autoinst-distri-opensuse/blame/master/lib/susedistribution.pm#L841 makes our live harder here. But I think this check should only apply to consoles which are using `tty`! 

So I added a exception https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12236/commits/1a7d051f144ade0b2712cb6d33e3b309dd0c36cd if the console is `is_serial_terminal()` 
@okurz WDYT?

**Review by pdostal — APPROVED** [review]
This now makes sense to me 👍 This will be very welcomed speedup 🚀
Let's just wait for @okurz as the author

**Comment by cfconrad** [comment]
> From what I understand, the TUNNELED consoles are all tty one.

Looks like I understood it wrong :(

So the trick is actually, that each console just call a `ssh -t sut` ... and the code expect to do the setup via `tunnel-console`.

File: lib/susedistribution.pm [hunk 2]
```diff
@@ -840,8 +840,12 @@ sub activate_console {
...
     if (get_var('TUNNELED') && $name !~ /tunnel/) {
         die "Console '$console' activated in TUNNEL mode activated but tunnel(s) are not yet initialized, use the 'tunnel' console and call 'setup_ssh_tunnels' first" unless get_var('_SSH_TUNNELS_INITIALIZED');
-        (get_var('PUBLIC_CLOUD')) ? ssh_interactive_join() : $self->script_run('ssh -t sut', 0);
-        ensure_user($user) unless (get_var('PUBLIC_CLOUD'));
+        if (get_var('PUBLIC_CLOUD')) {
+            ssh_interactive_join();
+        } else {
+            $self->script_run('ssh -t sut', 0);
+            ensure_user($user);
+        }
```
Comment (pdostal) [inline lib/susedistribution.pm:848]: Why this?

File: tests/publiccloud/download_repos.pm [hunk 3]
```diff
@@ -63,22 +63,22 @@ sub run {
                 record_soft_failure("Download failed (rc=$ret):\n$maintrepo");
                 script_run("echo 'Download failed for $maintrepo ...' >> ~/repos/qem_download_status.txt");
             } else {
-                assert_script_run("echo -en '# $maintrepo:\\n\\n' >> /tmp/repos.list.txt");
-                script_run("echo 'Downloaded $maintrepo: `du -hs $maintrepo`' >> ~/repos/qem_download_status.txt");
+                assert_script_run("echo -en '\\n" . ('#' x 80) . "\\n# $maintrepo:\\n' >> /tmp/repos.list.txt");
```
Comment (pdostal) [inline tests/publiccloud/download_repos.pm:66]: What does `('#' x 80)` mean?

File: lib/susedistribution.pm (hunk 2 above)
Comment (cfconrad) [reply lib/susedistribution.pm:848]: only readability -- I hope I didn't changed the logic, right?

File: tests/publiccloud/download_repos.pm (hunk 3 above)
Comment (cfconrad) [reply tests/publiccloud/download_repos.pm:66]: A line with `#`, see:
```
> perl -e 'print("#"x80 . "\n");'
################################################################################

File: lib/susedistribution.pm (hunk 2 above)
Comment (pdostal) [reply lib/susedistribution.pm:848]: Ou, my mistake.

Comment (cfconrad) [reply lib/susedistribution.pm:848]: What I actually wonder is, why we do not use `ensure_user()` within `PUBLIC_CLOUD`.

Comment (pdostal) [reply lib/susedistribution.pm:848]: I am not sure :/ Can you try to run `ensure_user($user)` without `PUBLIC_CLOUD`?

**Comment by cfconrad** [comment]
VR's with latest version:
https://openqa.suse.de/t5792577 - publiccloud_ltp_cve@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => passed
https://openqa.suse.de/t5792573 - mau-extratests-docker-publiccloud@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => passed
https://openqa.suse.de/t5792576 - publiccloud_img_proof@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => failed (expected failure -- not related to change)
https://openqa.suse.de/t5792575 - qem_publiccloud_img_proof@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => passed
https://openqa.suse.de/t5792574 - mau-extratests-publiccloud@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => passed

File: lib/susedistribution.pm [hunk 4]
```diff
@@ -840,7 +839,7 @@ sub activate_console {
     }
     if (get_var('TUNNELED') && $name !~ /tunnel/) {
         die "Console '$console' activated in TUNNEL mode activated but tunnel(s) are not yet initialized, use the 'tunnel' console and call 'setup_ssh_tunnels' first" unless get_var('_SSH_TUNNELS_INITIALIZED');
-        (get_var('PUBLIC_CLOUD')) ? ssh_interactive_join() : $self->script_run('ssh -t sut', 0);
+        $self->script_run('ssh -t sut', 0);
         ensure_user($user) unless (get_var('PUBLIC_CLOUD'));
```
Comment (cfconrad) [inline lib/susedistribution.pm:843]: I'm not sure about this line. I actually would expect that there is no difference for `PUBLIC_CLOUD` but 
`su - $user` give a login prompt again but the code afterwards isn't aware of it, or do I miss something?

Comment (grisu48) [reply lib/susedistribution.pm:843]: I'm not sure if `sut` is configured for publiccloud runs. But I share your concerns, this looks rather like a workaround than a necessity.

File: tests/publiccloud/download_repos.pm [hunk 5]
```diff
@@ -63,22 +63,22 @@ sub run {
...
+                assert_script_run("echo 'Downloaded $maintrepo:' \$(du -hs $parent | cut -f1) >> ~/repos/qem_download_status.txt");
                 if (script_run("ls $parent*.repo") == 0) {
                     assert_script_run("sed -i \"1 s/\\]/_\$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 4)]/\" $parent*.repo");
```
Comment (grisu48) [inline tests/publiccloud/download_repos.pm:69]: I find this line horrible. Perhaps we should refactor it in a future PR.

File: tests/publiccloud/ssh_interactive_start.pm [hunk 6]
```diff
@@ -16,17 +16,39 @@ use Mojo::Base 'publiccloud::ssh_interactive_init';
 use publiccloud::ssh_interactive;
 use testapi;
 use utils;
+use publiccloud::utils "select_host_console";
 
 sub run {
     my ($self, $args) = @_;
-    select_console 'tunnel-console';
+
+    # This ensure that we have used the setup console, even if no module was run before.
```
Comment (grisu48) [inline tests/publiccloud/ssh_interactive_start.pm:24]: Minor suggestion: What do you think to move lines 24-38 into it's own routine for better understanding? I'm thinking of just creating a `sub establish_tunnel` or something similar.

**Review by grisu48 — APPROVED** [review]
Only minor comments, LGTM otherwise.

I'd like to have a exhaustive set of verification runs here, because this contains some big changes - I can help you to schedule VR for QAM related test runs.

**Comment by cfconrad** [comment]
> Only minor comments, LGTM otherwise.
> 
> I'd like to have a exhaustive set of verification runs here, because this contains some big changes - I can help you to schedule VR for QAM related test runs.

Yes, I would appreciate your help! These https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12236#issuecomment-816155083 are the VR's I already run. I thought I cover most of it, but maybe we need to trigger different PC CSPs as we have some exceptions here and there.

File: tests/publiccloud/ssh_interactive_start.pm (hunk 6 above)
Comment (cfconrad) [reply tests/publiccloud/ssh_interactive_start.pm:24]: yes I think some further refactoring would be nice.

But we need to decide which direction. 
we have
* `lib:publiccloud:ssh_interactive_init` , which is a base class for:
  *  `tests/publiccloud/patch_and_reboot.pm`
  *  `tests/publiccloud/transfer_repos.pm`
  * `tests/publiccloud/register_system.pm`
  *  `tests/publiccloud/ssh_interactive_start.pm`
* `lib::publiccloud:ssh_interactive` which contain utility functions (each function is only called once!)
* and the connected tests:
  *  `tests/publiccloud/ssh_interactive_init.pm` (connect to PC instance and do preparations)
  *  `tests/publiccloud/ssh_interactive_start.pm` (final step to use `TUNNELED` mode of **all** consoles)
  *  `tests/publiccloud/ssh_interactive_end.pm` (cleanup and revert `TUNNELED` after all modules)

Things I like to cover:
* Don't use same name in test-module and lib
* Use a common baseclass for `ssh_interacive_(init|start|end).pm`
* Decide, if we wanna have the logic in some lib-function or directly visible in the test-module

Comment (cfconrad) [reply tests/publiccloud/ssh_interactive_start.pm:24]: And actually it would be nice to have a setup-console selection acrostic approach for `TUNNELED` behavior.
Something like
 * remember which console was already activated while `_TUNNELED_SETUP_IN_PROGRESS` is set
 * run over all consoles or do on select console the `ssh -t sut` when `_SSH_TUNNELS_INITIALIZED` is set

But I didn't found a good solution, thats why I would like to postpone it with this PR

@okurz  what do you think?

File: lib/susedistribution.pm (hunk 4 above)
Comment (cfconrad) [reply lib/susedistribution.pm:843]: yes we have, it is done here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/publiccloud/ssh_interactive_init.pm#L47

**Comment by grisu48** [comment]
Additional VRs

* [15-SP2-AZURE-BYOS-Updates mau-extratests](http://duck-norris.qam.suse.de/t5856)
* [15-SP2-AZURE-Basic-gen2-Updates mau-extratests](http://duck-norris.qam.suse.de/tests/5857)
* [15-SP2-EC2-ARM-Updates mau-extratests](http://duck-norris.qam.suse.de/tests/5858)
* [15-SP2-EC2-Updates mau-extratests](http://duck-norris.qam.suse.de/tests/5859)
* [15-SP2-GCE-Updates imgproof](http://duck-norris.qam.suse.de/tests/5860)
* [12-SP4-GCE-BYOS-Updates imgproof](http://duck-norris.qam.suse.de/tests/5861)

`openqa-mon -mfc 2 http://duck-norris.qam.suse.de/t5856..5861`

File: tests/publiccloud/download_repos.pm (hunk 5 above)
Comment (okurz) [reply tests/publiccloud/download_repos.pm:69]: What you could do is immediately create a ticket for the idea or maybe even better a (draft) PR to catch the idea as this is certainly out of scope for this PR

File: lib/publiccloud/ssh_interactive.pm [hunk 7]
```diff
@@ -40,18 +40,9 @@ sub ssh_interactive_tunnel {
 
     set_var('SERIALDEV_',               $serialdev);
     set_var('_SSH_TUNNELS_INITIALIZED', 1);
-}
-
-sub ssh_interactive_join {
```
Comment (okurz) [inline lib/publiccloud/ssh_interactive.pm:45]: Just leaving a method "interactive_leave" is not that explanatory. I would prefer to keep the balance of "join+leave".

File: lib/publiccloud/utils.pm [hunk 8]
```diff
@@ -20,19 +20,37 @@ use warnings;
...
+
+        opensusebasetest::select_serial_terminal();
+        ssh_interactive_leave();
+
+        select_console('tunnel-console', await_console => 0);
+        send_key "ctrl-c";
+        send_key "ret";
+
+        set_var('_SSH_TUNNELS_INITIALIZED', 0);
+        sleep 3;
```
Comment (okurz) [inline lib/publiccloud/utils.pm:48]: You are using `sleep()` or are simulating a sleep-like functionality. This is introducing the risk of either unnecessarily wasting time or just shifting race conditions and making them even harder to investigate. You *can* use sleep but there are only very limited cases where is the best solution (e.g. yield threads in low-level system code). Here you have the classical problem in software design of [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science)) . sleep should not be used in place of proper synchronization methods. https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/ is one of multiple references that explains this: "The original problem is likely a timing/synchronization issue, ignoring it by hiding it with Thread.Sleep is only going to delay the problem and make it occur in random, hard to reproduce ways." And if the problem does not occur at least time is wasted. https://testing.googleblog.com/2021/03/test-flakiness-one-of-main-challenges.html states that as well: "Do NOT add arbitrary delays as these can become flaky again over time and slow down the test unnecessarily."

We are waiting because some condition changes some time ... keyword(s) is/are *some time*. It is important to know the exact condition that the code should wait for and check for that instead of delaying program execution for an arbitrary amount of time. For openQA tests in general just try to do what you as a human user would also do. You would not look on your watch waiting for exactly 100s before you look at the screen again, right? :)

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

File: lib/publiccloud/utils.pm [hunk 9]
```diff
@@ -20,19 +20,37 @@ use warnings;
...
+
+    if ($tunneled && check_var('_SSH_TUNNELS_INITIALIZED', 1)) {
+        die("Called select_host_console but we are in TUNNELED mode") unless ($args{force});
+
+        opensusebasetest::select_serial_terminal();
+        ssh_interactive_leave();
+
+        select_console('tunnel-console', await_console => 0);
+        send_key "ctrl-c";
+        send_key "ret";
```
Comment (okurz) [inline lib/publiccloud/utils.pm:45]: please use single ticks by default

File: lib/publiccloud/utils.pm [hunk 10]
```diff
@@ -20,19 +20,37 @@ use warnings;
...
+
+        set_var('_SSH_TUNNELS_INITIALIZED', 0);
+        sleep 3;
+        save_screenshot;
     }
-    send_key "ctrl-c";
-    send_key "ret";
+    set_var('TUNNELED', 0) if ($tunneled);
```
Comment (okurz) [inline lib/publiccloud/utils.pm:51]: I suggest to leave out the parentheses around the bare variable:

```suggestion
    set_var('TUNNELED', 0) if $tunneled;
```

File: tests/publiccloud/ssh_interactive_start.pm [hunk 11]
```diff
@@ -16,17 +16,39 @@ use Mojo::Base 'publiccloud::ssh_interactive_init';
...
+    my $setup_console = current_console();
 
     # Establish the tunnel (it will stay active in foreground and occupy this console!)
+    select_console('tunnel-console');
     ssh_interactive_tunnel($args->{my_instance});
 
-    # Switch to root-console and SSH to the instance
-    # every other loaded test must stay in root-console
-    select_console 'root-console';
+    # Fix serial terminal to use TUNNELED stuff
```
Comment (okurz) [inline tests/publiccloud/ssh_interactive_start.pm:32]: Can you rephrase that comment to avoid "Fix" and "stuff" and say what the real issue is? :)

**Review by okurz — CHANGES_REQUESTED** [review]

**Review by okurz — CHANGES_REQUESTED** [review]
typo in commit message "select_serial_teriminal"

File: tests/publiccloud/download_repos.pm (hunk 5 above)
Comment (cfconrad) [reply tests/publiccloud/download_repos.pm:69]: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12268

File: lib/publiccloud/ssh_interactive.pm (hunk 7 above)
Comment (cfconrad) [reply lib/publiccloud/ssh_interactive.pm:45]: agree, but I would like to do the refactoring afterwards.
From this https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12236#discussion_r610430599 you can see that we have `start` which corresponding to `join` and `stop` which corresponds to `leave`.
For me it's a bit double, so refactoring it later will clean it up.

Comment (cfconrad) [reply lib/publiccloud/ssh_interactive.pm:45]: if you agree I would keep it in that PR

**Review by okurz — APPROVED** [review]

**Comment by grisu48** [comment]
@cfconrad  I've scheduled a new set of VRs

Additional VRs

* [15-SP2-AZURE-BYOS-Updates mau-extratests](http://duck-norris.qam.suse.de/t5863)
* [15-SP2-AZURE-Basic-gen2-Updates mau-extratests](http://duck-norris.qam.suse.de/tests/5864)
* [15-SP2-EC2-ARM-Updates mau-extratests](http://duck-norris.qam.suse.de/tests/5865)
* [15-SP2-EC2-Updates mau-extratests](http://duck-norris.qam.suse.de/tests/5866)
* [15-SP2-GCE-Updates imgproof](http://duck-norris.qam.suse.de/tests/5867)
* [12-SP4-GCE-BYOS-Updates imgproof](http://duck-norris.qam.suse.de/tests/5868)

`openqa-mon -mfc 2 http://duck-norris.qam.suse.de/t5863..5868`

**Comment by cfconrad** [comment]
The one which fails https://openqa.suse.de/tests/5802186#step/ssh_interactive_init/54, I think it is not related to this PR, right!?

But I guess duo to the changes of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12232 I need a further grep for `select_host_console()`!

/wip

**Comment by cfconrad** [comment]
VR's with latest changes:

https://openqa.suse.de/t5811025 - qem_publiccloud_img_proof@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => passed
https://openqa.suse.de/t5811022 - mau-extratests-publiccloud@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => softfailed
https://openqa.suse.de/t5811021 - mau-extratests-publiccloud@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => obsoleted
https://openqa.suse.de/t5811023 - mau-extratests-publiccloud@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => softfailed
https://openqa.suse.de/t5811026 - qem_publiccloud_img_proof@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => softfailed
https://openqa.suse.de/t5811024 - mau-extratests-publiccloud@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2@cfconrad/os-autoinst-distri-opensuse#publiccloud_dl_repos2 => running

File: tests/publiccloud/ssh_interactive_start.pm [hunk 12]
```diff
@@ -16,17 +16,40 @@ use Mojo::Base 'publiccloud::ssh_interactive_init';
...
 
-    # Switch to root-console and SSH to the instance
-    # every other loaded test must stay in root-console
-    select_console 'root-console';
+    # Enable ssh connection on setup console, this is done normally with the
+    # first activation hook in susedistribution:activate_console()
+    if ($setup_console !~ /tunnel/) {
+        select_console($setup_console);
+        script_run('ssh -t sut', timeout => 0);
+    }
```
Comment (pdostal) [inline tests/publiccloud/ssh_interactive_start.pm:37]: I didn't get this change.

File: tests/publiccloud/ssh_interactive_start.pm [hunk 13]
```diff
@@ -16,17 +16,40 @@ use Mojo::Base 'publiccloud::ssh_interactive_init';
...
+    # This ensure that we have used the setup console, even if no module was run before.
+    select_host_console();
+    my $setup_console = current_console();
 
     # Establish the tunnel (it will stay active in foreground and occupy this console!)
+    select_console('tunnel-console');
     ssh_interactive_tunnel($args->{my_instance});
 
-    # Switch to root-console and SSH to the instance
-    # every other loaded test must stay in root-console
```
Comment (pdostal) [inline tests/publiccloud/ssh_interactive_start.pm:28]: This line is not true.

File: tests/publiccloud/ssh_interactive_start.pm [hunk 14]
```diff
@@ -16,17 +16,40 @@ use Mojo::Base 'publiccloud::ssh_interactive_init';
...
+    assert_script_run('test -e /dev/' . get_var('SERIALDEV'));
+    assert_script_run('test $(id -un) == "root"');
+
+    select_console('user-console');
+    assert_script_run('test -e /dev/' . get_var('SERIALDEV'));
+    assert_script_run('test $(id -un) == "' . $testapi::username . '"');
+
+    $self->select_serial_terminal();
+    assert_script_run('test -e /dev/' . get_var('SERIALDEV'));
+    assert_script_run('test $(id -un) == "root"');
```
Comment (pdostal) [inline tests/publiccloud/ssh_interactive_start.pm:52]: I would move this to `tests/publiccloud/instance_overview.pm`.

File: tests/publiccloud/ssh_interactive_start.pm (hunk 12 above)
Comment (cfconrad) [reply tests/publiccloud/ssh_interactive_start.pm:37]: If we use the `tunnel` console, there is no need to do this `ssh -t sut` trick on the the setup_console. Cause it is already done by the fist activation of the console in this line: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/susedistribution.pm#L843

But if we use a different console (like the `root-virtio-terminal`) we need to setup this ssh connection later on.

File: tests/publiccloud/ssh_interactive_start.pm (hunk 14 above)
Comment (cfconrad) [reply tests/publiccloud/ssh_interactive_start.pm:52]: It's not related to the instance. The purpose of these lines are:

Check on `root-console`, `user-console` and `serial_teriminal (e.g. `root-virtio-terminal` ) that the ssh tunnel is setup properly.
Actually this is just a `sanity check for this `TUNNELED` behavior. We could remove it, but I found it handy so we are aware that it worked once.

File: tests/publiccloud/ssh_interactive_start.pm (hunk 13 above)
Comment (cfconrad) [reply tests/publiccloud/ssh_interactive_start.pm:28]: Don't know what to say. This is how I understood it, this line: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/susedistribution.pm#L843 does the trick on **every** console expect `tunnel`.

File: tests/publiccloud/ssh_interactive_start.pm (hunk 14 above)
Comment (pdostal) [reply tests/publiccloud/ssh_interactive_start.pm:52]: The `instance_overview` test module is also basically just sanity checking :)

File: tests/publiccloud/ssh_interactive_start.pm (hunk 13 above)
Comment (pdostal) [reply tests/publiccloud/ssh_interactive_start.pm:28]: Exactly. This was not a suggestion - this was just a statement that this line was not saying the truth.

File: tests/publiccloud/ssh_interactive_start.pm (hunk 12 above)
Comment (pdostal) [reply tests/publiccloud/ssh_interactive_start.pm:37]: I still don't get it. We setup the tunnel by setting `TUNNELED=1` and calling `select_console 'root-console';`.

**Review by pdostal — APPROVED** [review]
Thank you guys for all the research and verifications, this is another massive speed up! 🚀

File: tests/publiccloud/ssh_interactive_start.pm (hunk 6 above)
Comment (okurz) [reply tests/publiccloud/ssh_interactive_start.pm:24]: I guess it's merely the decision between keeping the "remote ssh tunnel" approach more explicit and hence different or streamline it as part of what we consider the normal console based approach for all cases. I am not sure who you would want it to look like. But feel welcome to go ahead :)
